lib/spack/spack/cmd/location.py

```diff
@@ -47,9 +47,13 @@ def setup_parser(subparser):
     directories.add_argument(
         '-S', '--stages', action='store_true',
         help="top level stage directory")
+    directories.add_argument(
+        '--source-dir', action='store_true',
```

Comment:
Should we add a short version here? Perhaps `-c`? I see that `-s` is already taken, but it feels weird that this is the only flag having only a long version.

Reply:
After discussion: -c is a bit unexpected, so instead there's no short version.